src/redux/wallet/actions.js

```diff
@@ -69,56 +70,48 @@ export const updateTIMEBalance = () => async (dispatch) => {
 }
 
 export const updateTIMEDeposit = () => async (dispatch) => {
+  dispatch({type: WALLET_TIME_DEPOSIT_FETCH})
   const dao = await contractsManagerDAO.getTIMEHolderDAO()
   const deposit = await dao.getAccountDepositBalance(ls.getAccount())
   dispatch({type: WALLET_TIME_DEPOSIT, deposit})
 }
 
-// TODO This is only for test purposes
-// export const requireTIME = () => async (dispatch) => {
-//   dispatch(hideModal())
-//   dispatch(balanceFetch(TIME))
-//   try {
-//     await TokenContractsDAO.requireTIME()
-//     const token = await contractsManagerDAO.getTIMEDAO()
-//     return dispatch(updateBalance(token))
-//   } catch (e) {
-//     dispatch(balanceFetch(TIME))
-//   }
-// }
+export const requireTIME = () => async (dispatch) => {
+  try {
+    const dao = await contractsManagerDAO.getAssetDonator()
+    await dao.requireTIME()
+  } catch (e) {
+    dispatch(showAlertModal({title: 'Require TIME error', message: e.message}))
+  }
+  dispatch(updateTIMEBalance())
+}
 
 export const depositTIME = (amount) => async (dispatch) => {
-  dispatch(hideModal())
-  dispatch(balanceFetch(TIME))
   try {
     const dao = await contractsManagerDAO.getTIMEHolderDAO()
     const result = await dao.deposit(amount)
-    dispatch(updateTIMEBalance())
-    if (result) {
-      dispatch(updateTIMEDeposit())
-    } else {
+    if (!result) {
       dispatch(showAlertModal({title: 'Deposit TIME error', message: 'Insufficient funds.'}))
     }
   } catch (e) {
     dispatch(showAlertModal({title: 'Deposit TIME error', message: e.message}))
```

Comment:
txEnd will notify user about any tx error, remove this line
catch of tx code should contain only rollback behaviour